src/rust/catnap/futures/pop.rs

```diff
@@ -78,34 +81,40 @@ impl Future for PopFuture {
     /// Polls the target [PopFuture].
     fn poll(self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Self::Output> {
         let self_: &mut PopFuture = self.get_mut();
-        let mut bytes: [u8; POP_SIZE] = [0; POP_SIZE];
-        match socket::recvfrom::<SockaddrStorage>(self_.fd, &mut bytes[..]) {
+        match unsafe {
+            let mut addrlen: libc::socklen_t = 0;
+            libc::recvfrom(
+                self_.fd,
+                (self_.buf.as_mut_ptr() as *mut u8) as *mut libc::c_void,
+                self_.buf.len(),
+                libc::MSG_DONTWAIT,
```

Comment:
Why this change?  Shouldn't this fd already be set_nonblock?

Reply:
When creating the socket the `set_nonblock()` call may not succeed. In that case the socket is still created, but it would be a blocking one. By supplying the `libc::MSG_DONTWAIT` we are doing our best to make any asynchronous operation in a blocking socket.

Reply:
Okay.